components/omega/doc/design/Logging.md

```diff
@@ -0,0 +1,109 @@
+
+### 4.1 Data types and parameters
+
+#### 4.1.1 Parameters 
+
+The software design will implement a global switch for setting the minimum log severity and message format using predefined global C++ defines. The severity level can be set at compile-time using -D LOG\_LEVEL=<level>, where <level> can be one of the predefined levels: "TRACE", "DEBUG", "INFO", "WARN", "ERROR", or "CRITICAL", where the severity increases from TRACE to CRITICAL. Log messages that have severity equal to or higher than the specified level will be stored in the log destination.
+
+Furthermore, the message format can be set using -D LOG\_PATTERN=<pattern> at compile time, where <pattern> represents a format in the form of %flag, similar to the strftime function. This pattern defines the layout of the log message.
+
+#### 4.1.2 Class/structs/data types
+
+No public data type is necessary for the logging system.
+
+### 4.2 Methods
+
+The OMEGA logging system includes multiple macros that allow for logging at various severity levels. These macros guarantee that log messages with severity levels lower than the specified LOG\_LEVEL will not be compiled.
+
+#### 4.2.1 Log write
+
+The OMEGA logging system includes macros that allow the user to write log messages with different severity levels. These macros follow the naming convention LOG_<severity level>. For instance, the following macro can be used to write an informational log message:
+
+    LOG_INFO("log message");
+
+The OMEGA logging system provides the following macros for writing log messages with different severity levels:
+
+1. LOG\_CRITICAL: Used to indicate a critical error or issue that requires immediate attention and could potentially cause data loss or application failure.
+2. LOG\_ERROR: Used to indicate an error or issue that could cause application instability or incorrect behavior.
+3. LOG\_WARN: Used to indicate a warning or potential issue that should be addressed but does not necessarily require immediate attention.
+4. LOG\_INFO: Used to provide general informational messages about the application's operation or progress.
+5. LOG\_DEBUG: Used to provide detailed debugging information for developers to track down issues or analyze behavior.
+6. LOG\_TRACE: Used to provide the most detailed level of information, including function calls and variable values, and is typically used for deep analysis and debugging.
+
+In order to enhance the logging capabilities, OMEGA logging system provides each macro with an optional argument of an Omega data type. If the data type is recognized, it will be formatted automatically by the system and appended in the appropriate format to the log message. As an example, the following macro demonstrates how an argument of type Array1DR8 can be appended to a log message:
```

Comment:
```suggestion
In order to enhance the logging capabilities, OMEGA logging system provides each macro with an optional argument of an Omega data type. If the data type is recognized, it will be formatted automatically by the system and appended in the appropriate format to the log message. As an example, the following macro demonstrates how an argument of type `Array1DR8` can be appended to a log message:
```